sklearn/decomposition/incremental_pca.py

```diff
@@ -210,11 +210,19 @@ def partial_fit(self, X, y=None, check_input=True):
             self.components_ = None
 
         if self.n_components is None:
-            self.n_components_ = n_features
+            if self.components_ is None:
```

Comment:
Can you add regression tests for these? I guess if ``n_features < n_samples`` we had an error earlier?